### PR TITLE
allow cjk characters in commit messages, branch names, etc.

### PR DESCRIPTION
**Pull request by frostbane**
I work in Japan and branches, commits, authors, etc. containing Japanese characters become unreadable.

Then I fixed the repository link in the main menu.

### CODE REVIEW

File: Source/Source.php [hunk 1]
```diff
@@ -141,7 +141,7 @@ function menu_main() {
 		$t_menu_options = array();
 
 		if ( plugin_config_get( 'show_repo_link' ) ) {
-			$t_page = plugin_page( 'index', false, 'Source' );
+			$t_page = plugin_page( 'index', true, 'Source' );
```
Comment (dregad) [inline Source/Source.php]: This makes the URL returned by plugin_page() a full URI with protocol, host, etc.
I do not understand why it is necessary to change this in the context of allowing CJK chars, and furthermore AFAICT it works just fine with the current value (short URL with only the path).

File: SourceBitBucket/SourceBitBucket.php [hunk 2]
```diff
@@ -246,9 +250,12 @@ public function import_full( $p_repo, $p_use_cache = true ) {
...
 				}
 			}
-			if( $p_use_cache ) foreach ( $t_commits as $t_commit_id ) $this->load_all_commits( $p_repo, $t_commit_id );
+			if( $p_use_cache )
+				foreach ( $t_commits as $t_commit_id )
+					$this->load_all_commits( $p_repo, $t_commit_id );
 
-			$t_changesets = array_merge( $t_changesets, $this->import_commits( $p_repo, $t_commits, $t_branch ) );
+			$t_changesets = array_merge( $t_changesets,
+										 $this->import_commits( $p_repo, $t_commits, $t_branch ) );
```
Comment (dregad) [inline SourceBitBucket/SourceBitBucket.php]: This seems to be only a whitespace change... What is it for ?

**Comment by dregad** [comment]
I'm also a bit surprised that you would need to use utf8_decode(), since json data is expected to be UTF8, not ISO-8859-1, and the DB should be UTF8 as well.

**Comment by frostbane** [comment]
Here is a screenshot of a repository with Japanese commits.

screenshot_2016-11-13-000359

File: Source/Source.php (hunk 1 above)
Comment (dregad) [inline Source/Source.php]: >  I'll move it to another pull request. 

Thanks. I'll wait for you to do that, then we can follow up on this discussion there. 

The fact that it works for me as it is on several instances (including http://mantisbt.org/bugs) leads me to believe that it must be something related to your environment's setup, and I'd like to understand what it is.

**Comment by dregad** [comment]
Can you point me to a public bitbucket repo that I can use for testing purposes, to try and replicate the problem ?

**Comment by dregad** [comment]
@frostbane what is the status on this PR ?

File: Source/Source.php (hunk 1 above)
Comment (dregad) [inline Source/Source.php]: This was actually fixed in 1491ba13b288d1b8ac18d3025e3db29e62265cb5, see also http://www.mantisbt.org/bugs/view.php?id=22107

**Comment by dregad** [comment]
Without feedback from you, I'll close this PR; feel free to reopen it if you can provide information to help reproduce the problem.

**Pull request closed without merging**